### PR TITLE
Com templates file permissions

### PR DESCRIPTION
**Pull request by jsubri**
Trying to update a template in an unwritable directory COM_TEMPLATES_FILE_PERMISSIONS message is not displayed.

com_templates_file_permissions-before

after the patch

com_templates_file_permissions-after

### CODE REVIEW

**Comment by infograf768** [comment]
This is not language friendly: the variable should be displayed depending on the language.
Patch should be imho:
```
diff --git a/administrator/components/com_templates/models/template.php b/administrator/components/com_templates/models/template.php
index b88b615..d75b735 100644
--- a/administrator/components/com_templates/models/template.php
+++ b/administrator/components/com_templates/models/template.php
@@ -494,5 +494,5 @@
 		{
 			$app->enqueueMessage(JText::_('COM_TEMPLATES_ERROR_SOURCE_FILE_NOT_WRITABLE'), 'warning');
-			$app->enqueueMessage(JText::_('COM_TEMPLATES_FILE_PERMISSIONS' . JPath::getPermissions($filePath)), 'warning');
+			$app->enqueueMessage(JText::sprintf('COM_TEMPLATES_FILE_PERMISSIONS', JPath::getPermissions($filePath)), 'warning');
 
 			if (!JPath::isOwner($filePath))
diff --git a/administrator/language/en-GB/en-GB.com_templates.ini b/administrator/language/en-GB/en-GB.com_templates.ini
index 9591641..a00c4a1 100644
--- a/administrator/language/en-GB/en-GB.com_templates.ini
+++ b/administrator/language/en-GB/en-GB.com_templates.ini
@@ -117,5 +117,5 @@
 COM_TEMPLATES_FILE_INFO="File Information"
 COM_TEMPLATES_FILE_NAME="File Name"
-COM_TEMPLATES_FILE_PERMISSIONS="The File Permissions are "
+COM_TEMPLATES_FILE_PERMISSIONS="The File Permissions are %s"
 COM_TEMPLATES_FILE_RENAME_ERROR="An error occurred renaming the file."
 COM_TEMPLATES_FILE_RENAME_SUCCESS="File renamed successfully."
```

**Comment by jsubri** [comment]
Implemented suggestions, thank you.

**Comment by infograf768** [comment]
:+1:

**Comment by brianteeman** [comment]
Setting the protostar folder to unwritable I have only been able to display the error message and not the warning<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by jsubri** [comment]
@Brianteeman, Thank you for testing. 
You are not wrong, the Error message is displayed when the directory is unwritable, Warning appears at the file level.

Full test:
set 777 recursive to /administrator/templates/hathor 
set 755 to administrator/templates/hathor.dir 
set  755 or 644 to administrator/templates/hathor/component.php
updating administrator/templates/hathor/component.php you should get both Error and Warning

line 496 should have a comma and not a dot to get the message evaluated.
Tks

**Comment by brianteeman** [comment]
Sorry I still cant replicate it with those instructions

On 24 May 2015 at 16:50, Jean-Luc Subri <notifications@github.com> wrote:

> @Brianteeman <https://github.com/Brianteeman>, Thank you for testing.
> You are not wrong, the Error message is displayed when the directory is
> unwritable, Warning appears at the file level.
>
> Full test:
> set 777 recursive to /administrator/templates/hathor
> set 755 to administrator/templates/hathor.dir
> set 755 or 644 to administrator/templates/hathor/component.php
> updating administrator/templates/hathor/component.php you should get both
> Error and Warning
>
> line 496 should have a comma and not a dot to get the message evaluated.
> Tks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6609#issuecomment-105028571>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

File: administrator/components/com_templates/models/template.php [hunk 1]
```diff
@@ -493,7 +493,7 @@ public function save($data)
 		if (!is_writable($filePath))
 		{
 			$app->enqueueMessage(JText::_('COM_TEMPLATES_ERROR_SOURCE_FILE_NOT_WRITABLE'), 'warning');
-			$app->enqueueMessage(JText::_('COM_TEMPLATES_FILE_PERMISSIONS' . JPath::getPermissions($filePath)), 'warning');
```
Comment (smz) [inline administrator/components/com_templates/models/template.php]: Sorry, but wouldn't adding a blank to the string constant be enough, without resorting to sprintf and changing the language string?
```
$app->enqueueMessage(JText::_('COM_TEMPLATES_FILE_PERMISSIONS ' . JPath::getPermissions($filePath)), 'warning');
```

Comment (smz) [inline administrator/components/com_templates/models/template.php]: or:

```
$app->enqueueMessage(JText::_('COM_TEMPLATES_FILE_PERMISSIONS') . ' ' . JPath::getPermissions($filePath), 'warning');
```

Comment (sovainfo) [inline administrator/components/com_templates/models/template.php]: For translating and overriding the message it is better to use sprintf.

Comment (smz) [inline administrator/components/com_templates/models/template.php]: Got it! Thanks!

Comment (jsubri) [inline administrator/components/com_templates/models/template.php]: Thank you for your comments, with english OS I had the same output with both JText:_ and JTest:sprintf . sprintf seems more appropriate, draw back is the impact on the langue string. 
@test is it RTC based on test or review(s)? probably late for 3.4.2

**Comment by Bloggerschmidt** [comment]
I'm sorry, but I can't reproduced this issue with the test instructions above, so hathor.dir file or dir doesn't exists. Hm.

**Comment by svom** [comment]
I'm not able to reproduce this issue with the instructions above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by designbengel** [comment]
Can´t reproduce the issue. Got the message: Error
The template folder is not writable. Some features may not work.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by jsubri** [comment]
Thank you for testing. 
Reading your comments, you are all receiving the Error message but none of you is entering the condition on line  493 => "if (!is_writable($filePath))"

With my hoster, the customers are receiving user/group "customer-xx/webusers" while Apache is running under "webserverd/webserverd".

Might be the best way to get the Warning is to comment out the line 493 and testing again!
Best
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by RickR2H** [comment]
I could reproduce the error by using this procedure:
Set the folder permissons of the Protostar template to 755 (Normal Apache rights)
Open the protostar template and edit a file from the internal template editor and make a small change. Don't press save yet!
Next, change the folder permissions of the protostar template to 444
Now go back and press Save to store the changes
The error will now occur.

The only thing is that the folder rights are only dashes like: ---------
screen shot 2016-04-15 at 08 01 08

But the translations is fixed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 6888366d4dcc90191f86d392402e5a351e5928d0<br /><br />See my previous post...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by designbengel** [comment]
Before: Source file not writable.
COM_TEMPLATES_FILE_PERMISSIONSr--r--r--
Could not save data. Error:

After: Source file not writable.
The File Permissions are r--r--r--
Could not save data. Error:

But isn´t there something missing after "Error:" ?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by brianteeman** [comment]
yes something is missing

On 15 April 2016 at 14:12, designbengel <notifications@github.com> wrote:

> Before: Source file not writable.
> COM_TEMPLATES_FILE_PERMISSIONSr--r--r--
> Could not save data. Error:
>
> After: Source file not writable.
> The File Permissions are r--r--r--
> Could not save data. Error:
>
> But isn´t there something missing after "Error:" ?
> ------------------------------
> This comment was created with the J!Tracker Application
> <https://github.com/joomla/jissues> at issues.joomla.org/joomla-cms/6609
> <https://issues.joomla.org/tracker/joomla-cms/6609>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6609#issuecomment-210459948>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by smz** [comment]
Nitpicking: either this is an **error** or this is a **warning**: in both cases I think wording and alert class (_color_) should be consistent.

**Comment by henkrijneveld** [comment]
I have tested this item :red_circle: unsuccessfully on 6888366d4dcc90191f86d392402e5a351e5928d0<br /><br />BEFORE:
Warning
Source file not writable.
COM_TEMPLATES_FILE_PERMISSIONSr--r--r--
Could not save data. Error:

AFTER
Warning
Source file not writable.
The File Permissions are r--r--r--
Could not save data. Error:

apart from the upper to lowercase no difference. more work needed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by conconnl** [comment]
I have tested this item :white_check_mark: successfully on 6888366d4dcc90191f86d392402e5a351e5928d0<br /><br />Tested the patch successfully.
The language string is working again.

Before and After the patch the file permissions were written like: ------
Before and After the patch the Error: is empty

There are other errors but not because of the patch<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by pjasmits** [comment]
Found some missing strings in the file: en-GB.com_templates.ini

COM_TEMPLATES_STYLES_PAGES_ALL="Default for all pages"
COM_TEMPLATES_STYLES_PAGES_NONE="Not assigned"
COM_TEMPLATES_MANAGER_STYLES_SITE="Templates: Styles (Site)"
COM_TEMPLATES_HEADING_PAGES="Pages"
COM_TEMPLATES_MANAGER_TEMPLATES_SITE="Templates: Templates (Site)"
COM_TEMPLATES_HEADING_ASSIGNED="Assigned"
COM_TEMPLATES_HEADING_DEFAULT="Default"
COM_TEMPLATES_HEADING_DEFAULT_ASC="Default ascending"
COM_TEMPLATES_HEADING_DEFAULT_DESC="Default descending"
COM_TEMPLATES_HEADING_IMAGE="Image"
COM_TEMPLATES_HEADING_LOCATION_ASC="Location ascending"
COM_TEMPLATES_HEADING_LOCATION_DESC="Location descending"
COM_TEMPLATES_HEADING_PAGES="Pages"
COM_TEMPLATES_HEADING_STYLE="Style"
COM_TEMPLATES_HEADING_STYLE_ASC="Style ascending"
COM_TEMPLATES_HEADING_STYLE_DESC="Style descending"
COM_TEMPLATES_HEADING_TEMPLATE="Template"
COM_TEMPLATES_HEADING_TEMPLATE_ASC="Template ascending"
COM_TEMPLATES_HEADING_TEMPLATE_DESC="Template descending"
COM_TEMPLATES_STYLES_PAGES_SELECTED="Assigned on %s pages"

screen shot 2016-06-25 at 06 29 35

Test is successful some work needs to be done in the file en-GB.com_templates.ini
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by pjasmits** [comment]
I have tested this item :red_circle: unsuccessfully on 6888366d4dcc90191f86d392402e5a351e5928d0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by pjasmits** [comment]
I have tested this item :white_check_mark: successfully on 6888366d4dcc90191f86d392402e5a351e5928d0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by zero-24** [comment]
@joomla-cms-bot please help me :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6609">issues.joomla.org/joomla-cms/6609</a>.</sub>

**Comment by infograf768** [comment]
@pjasmits

These strings are present in staging/3.6.0 beta

**Comment by jsubri** [comment]
Thank you all for your tests.
Best

**Comment by zero-24** [comment]
@roland-d can You re-add the milestone? I have called the bot too fast yesterday ;)

**Comment by zero-24** [comment]
@joomla-cms-bot did i something wrong? Why you don't help me?

**Pull request merged**